ts/features/wallet/onboarding/bancomat/screens/SearchBankScreen.tsx

```diff
@@ -1,18 +1,137 @@
-import { View } from "native-base";
+import { Content, Input, Item, Label, Text, View } from "native-base";
 import * as React from "react";
+import { FlatList, ListRenderItemInfo, SafeAreaView } from "react-native";
 import { connect } from "react-redux";
 import { Dispatch } from "redux";
+import { Link } from "../../../../../components/core/typography/Link";
+import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
 import { GlobalState } from "../../../../../store/reducers/types";
+import IconFont from "../../../../../components/ui/IconFont";
+import { BankPreviewItem } from "../components/BankPreviewItems";
+import { abis } from "../mock/mockData";
+import {
+  cancelButtonProps,
+  confirmButtonProps
+} from "../../../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import I18n from "../../../../../i18n";
+import { navigateBack } from "../../../../../store/actions/navigation";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import { withLightModalContext } from "../../../../../components/helpers/withLightModalContext";
+import { LightModalContextInterface } from "../../../../../components/ui/LightModal";
+import TosBonusComponent from "../../../../bonus/bonusVacanze/components/TosBonusComponent";
 
+type Props = LightModalContextInterface &
+  ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
 /**
  * This screen allows the user to choose a specific bank to search for their Bancomat.
  * the user can also choose not to specify any bank and search for all Bancomat in his name
  * @constructor
  */
-export const SearchBankScreen: React.FunctionComponent = () => <View />;
+const SearchBankScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const [searchText, setSearchText] = React.useState("");
+  const [filteredList, setFilteredList] = React.useState(
+    [] as typeof props.bankList
+  );
 
-const mapDispatchToProps = (_: Dispatch) => ({});
+  const [isSearchStarted, setIsSearchStarted] = React.useState(false);
 
-const mapStateToProps = (_: GlobalState) => ({});
+  const renderListItem = (isList: boolean) => (
+    info: ListRenderItemInfo<any>
+  ) => (
+    <BankPreviewItem
+      bank={info.item}
+      inList={isList}
+      onPress={props.onItemPress}
+    />
+  );
 
-export default connect(mapStateToProps, mapDispatchToProps)(SearchBankScreen);
+  const handleFilter = (text: string) => {
+    setIsSearchStarted(true);
+    const resultList = props.bankList.filter(
+      bank =>
+        bank.abi.toLowerCase().indexOf(text.toLowerCase()) > -1 ||
+        bank.name.toLowerCase().indexOf(text.toLowerCase()) > -1
+    );
+    setFilteredList(resultList);
```

Comment:
It should be sorted by name, I guess
@fabriziofff do you confirm?

Reply:
@Undermaken @fabriziofff does the sorting comes from API result or do we need to sort on the client too?

Reply:
I don't now.
API could change. If we have to display abi sorted, I include a client sorting anyway

Reply:
Solved on 0cb9afb